src/components/study-container.js

```diff
@@ -594,17 +594,28 @@ export function StudyContainer({ view, onChangeTab }) {
                 }
                 // updating data related to impacted substations
                 if (substationsIds?.length > 0) {
+                    const previousCurrentNode = currentNode;
                     console.info('Reload network equipments');
-                    network.reloadImpactedSubstationsEquipments(
-                        studyUuid,
-                        currentNodeRef.current,
-                        substationsIds
-                    );
-                    dispatch(setUpdatedSubstationsIds(substationsIds));
+                    network
+                        .reloadImpactedSubstationsEquipments(
+                            studyUuid,
+                            currentNodeRef.current,
+                            substationsIds
+                        )
+                        .then((values) => {
+                            if (
+                                previousCurrentNode === currentNodeRef.current
+                            ) {
+                                network.updateNetworkEquipments(values);
+                                dispatch(
+                                    setUpdatedSubstationsIds(substationsIds)
+                                );
+                            }
+                        });
                 }
             }
         }
-    }, [studyUpdatedForce, network, studyUuid, dispatch]);
+    }, [studyUpdatedForce, network, studyUuid, dispatch, currentNode]);
```

Comment:
to avoid adding a new useEffect dependency you could use `currentNodeRef.current` instead of `currentNode`
in
```
const previousCurrentNode = currentNodeRef.current;
```
then it is also miss-named, maybe something like `currentNodeAtRelaodCalling` is a better name

Reply:
Done